controllers/rhmi/rhmi_controller.go

```diff
@@ -1499,3 +1618,34 @@ func getInstallation() (*rhmiv1alpha1.RHMI, error) {
 		},
 	}, nil
 }
+
+func formatAlerts(alerts []struct {
+	Labels models.LabelSet `json:"labels"`
+	State  string          `json:"state"`
+}) resources.AlertMetrics {
+	alertMetrics := resources.AlertMetrics{}
+
+	for _, alert := range alerts {
+		if alert.Labels["alertname"] == "DeadMansSwitch" {
+			// DeadManSwitch alert is always fire and only a problem if not firing.
+			// Skipping the inclusion to have a more accurate metric.
+			continue
+		}
+		if !alertMetrics.Contains(alert) {
+			alertMetrics.Alerts = append(alertMetrics.Alerts, resources.AlertMetric{
+				Name:     alert.Labels["alertname"],
+				Severity: alert.Labels["severity"],
+				State:    alert.State,
+				Value:    0,
+			})
+		}
+
+		for index := range alertMetrics.Alerts {
+			if alertMetrics.Alerts[index].Contains(alert) {
+				alertMetrics.Alerts[index].Value++
+			}
+		}
```

Comment:
The initial value of the alert metric is zero for the alert. This loop is increasing the alert count by 1. We are not setting the value to 1. For example `ThreeScaleContainerHighMemory` alert is firing for two pods, in the metric there will only be one instance for `ThreeScaleContainerHighMemory` but have a count of 2.

```
rhoam_alerts{alert="ThreeScaleContainerHighMemory",severity="info",state="firing"} 2
```

This can not go out side the current loop as we need to iterate over both `alerts` and `alertMetrics` to find the matches. I do agree the nested loops are not great but are needed. Adding a `break` into the `if` on 1644 will help but won't completely remove the need.

Reply:
Thanks for the explanation. So the idea is the value of the metric shows how many instances of the alert is firing?

Reply:
yes, the idea is to get a summary of the alerts and if SRE wants to see the alerts that are firing they would check in the cluster with the alerts.

Reply:
Explained during code review. A rewrite of logic was asked for.